src/librustc_interface/passes.rs

```diff
@@ -939,7 +940,10 @@ fn analysis<'tcx>(
 
     // Abort so we don't try to construct MIR with liveness errors.
     // We also won't want to continue with errors from rvalue promotion
```

Comment:
well, we actually want to, but that's not really a problem of this PR. Have you checked how bad the fallout is if we remove this `abort_if_errors`?

Reply:
I believe that there were some useless errors coming from const evaluation, but those might have come from removing the later check at https://github.com/rust-lang/rust/pull/59903/files/13a05a27e9a5747cad090b1670c0a6b0baa624b9#diff-24c5c945888bb0d041e769bfb852de6cR976

Reply:
Can you check? I'd rather remove it entirely. Const eval should not evaluate MIR with typeck errors or borrowck errors.

Reply:
@oli-obk most of the extra errors are not too bad, but `src/test/ui/consts/match_ice.rs` starts ICEing again. I'll need to [dig deeper to fix the ICE and would like to remove the duplicated errors/warnings](https://github.com/rust-lang/rust/commit/8395dbd6dc0dd61c7ebb50a6afebb4d29c450a26) before merging that change, but it is certainly doable.

Reply:
Hmm... Those look like deeper issues to me where some things should be querified more or just produce dummy values in case of errors. Please open an issue linking to the fallout of https://github.com/rust-lang/rust/commit/8395dbd6dc0dd61c7ebb50a6afebb4d29c450a26 and mentioning the problems.